internal/discovery/dependentdiscovery.go

```diff
@@ -339,7 +339,8 @@ func (dd *DependentDiscovery) DiscoverDependents(
 
 				candidate.AddDependency(c)
 
-				if dep == target.Path() {
+				// Use resolved paths for comparison to handle symlinks (e.g., macOS /var -> /private/var)
+				if resolvePath(dep) == resolvePath(target.Path()) {
```

Comment:
nit: I wonder if it'd be reasonable at some point to do this path sanitization consistently closer to parse time rather than sprinkled throughout when we compare things 🤔 

this would probably require a universal "display vs execution" concept for all paths everywhere